internal/utils/msa/msa.go

```diff
@@ -0,0 +1,120 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// Not sure if this is idiomatic go, but I'm trying to make it easier
+// to work with untyped maps such as might be created by unmarshalling
+// arbitrary JSON data
+
+// Short for MapStringAny
+package msa
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type MapStringAny map[string]any
+
+// Create a MapStringAny from a plain map[string]any
+func FromMap(m map[string]any) MapStringAny {
+	return MapStringAny(m)
+}
+
+// Create a MapStringAny by unmarshaling json bytes
+func FromJsonBytes(bytes []byte) (MapStringAny, error) {
+	var m MapStringAny
+	err := json.Unmarshal(bytes, &m)
+	return m, err
+}
+
+// Create a MapStringAny by unmarshaling a json string
+func FromJson(str string) (MapStringAny, error) {
+	return FromJsonBytes([]byte(str))
+}
+
+// Convert back to a plain map[string]any
```

Comment:
Changing the resulting `map[string]any` will mutate `MapStringAny` might be unexpected, perhaps add that to the docs?

Reply:
Or... perhaps we could make a deep clone to prevent that mutation.

Reply:
See latest revision, top commit.